### PR TITLE
55 cutadapt

### PR DESCRIPTION
**Pull request by dhoogest**

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -562,6 +605,7 @@ if(params.containsKey("bidirectional") && params.bidirectional){
 
 process join_counts {
     input:
+        file("raw.csv") from raw_counts_concat
```
Comment (nhoffman) [inline main.nf]: Are cutadapt counts making it into here? i don't see a reference to ``cutadapt_counts_concat``, which I assume would hold the input file.

Comment (dhoogest) [reply main.nf]: No not yet - I haven't figured out how to pack/unpack a channel of files ... have some nextflow learnin' to do.  Alternatively could just define `cutadapt_counts_concat` as empty in the null condition.

Comment (dhoogest) [reply main.nf]: Okay not ideal, but after running out of patience trying to figure out how to concat then expand a non-deterministic list of counts_concat channels I went with the approach seen [here](https://github.com/nhoffman/dada2-nf/pull/58/commits/51dd38bc82fb66184c51440e33a25b11b56d848e). Basically defines an empty `cutadapt_counts_concat` channel and then tests channel content to set the join command. Happy to have this adjusted if either @nhoffman or @crosenth knows how to better achieve.

Comment (dhoogest) [reply main.nf]: Oops - please hold, think the logic is backward.

**Comment by dhoogest** [comment]
@nhoffman okay I think this branch is functionally complete now, although could be likely improved a bit in terms of how counts files are packed/unpacked. I have not added specific tests to inspect/verify the adaptor trimming, but would note that the `params-its.json` is run as part of the automated testing, and therefore both the 'with cutadapt_params' and 'without' conditions are currently included.

File: main.nf [hunk 2]
```diff
@@ -118,6 +118,43 @@ process plot_quality {
...
+    process cutadapt_counts_concat {
+
+    input:
+        file("*.cutadapt.tsv") from cutadapt_log.collect()
+
+    output:
+        file("cutadapt_counts.csv") into cutadapt_counts_concat
+
+    """
+    stack_cutadapt_counts.sh *.cutadapt.tsv | sed -e 's/out_reads/cutadapt/' | xsv select -d '\t' sampleid,cutadapt  > cutadapt_counts.csv
```
Comment (nhoffman) [inline main.nf:150]: Is there a reason not to move all of the necessary processing into ``stack_cutadapt_counts.sh`` and write directly to an output file?

Comment (dhoogest) [reply main.nf:150]: No -  I'm weak enough in bash that I didn't want to try to do the column selection without the help of xsv is the main reason I wrote it this way. I guess the command above could happen in the bash script directly (moving `sed` and `xsv` into the bash script), but not sure if this preferable to leaving commands in the main.nf.

Comment (nhoffman) [reply main.nf:150]: Ok, not important to make any changes now. May revisit later.

**Pull request merged**